public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -145,13 +145,18 @@ export class AppChromeService {
       return entries;
     }
 
-    let lastEntry = entries[0];
-    if (!lastEntry || lastEntry.name !== newPageNav.text) {
-      lastEntry = { name: newPageNav.text, views: [], breadcrumbs, time: Date.now(), url: window.location.href };
-    }
-    if (lastEntry !== entries[0]) {
-      entries = [lastEntry, ...entries];
+    const lastEntry = entries[0];
+    const newEntry = { name: newPageNav.text, views: [], breadcrumbs, time: Date.now(), url: window.location.href };
+    const isSameUrl = newEntry.url === lastEntry.url;
```

Comment:
just realised we need an extra check here, since `lastEntry` could be `undefined`
```suggestion
    const isSameUrl = lastEntry && (newEntry.url === lastEntry.url);
```